D2L.Hypermedia.Siren/SirenField.cs

```diff
@@ -22,11 +23,29 @@ public class SirenField : ISirenField {
 		) {
 			m_name = name;
 			m_class = @class ?? new string[0];
-			m_type = type;
+			m_type = ValidateType( type );
 			m_value = value;
 			m_title = title;
 		}
 
+		private string ValidateType( string type ) {
+			if( type == null ) {
+				return null;
+			}
+
+			type = type.ToLowerInvariant();
+
+			PropertyInfo[] properties = typeof( SirenFieldType ).GetProperties();

```

Comment:
Usage of that seems like it would be a bit awkward, though - in particular, you wouldn't be able to use code completion :(

Maybe just a backing `IDictionary<string, string>` within `SirenFieldType` that matches the properties, though? I agree, the reflection is a little... square peg, round hole.

Reply:
Another option would be an Enum for the keys (for code completion) and then a collection of <string,Enum> tuples?

Reply:
Aha! That may be the ticket - I'll give that a shot and see what's what :)